code/addons/toolbars/src/components/ToolbarMenuList.tsx

```diff
@@ -18,11 +19,12 @@ export const ToolbarMenuList: FC<ToolbarMenuListProps> = withKeyboardCycle(
     description,
     toolbar: { icon: _icon, items, title: _title, preventDynamicIcon, dynamicTitle },
   }) => {
-    const [globals, updateGlobals] = useGlobals();
+    const [globals, updateGlobals, userGlobals] = useGlobals();
     const [isTooltipVisible, setIsTooltipVisible] = useState(false);
 
     const currentValue = globals[id];
     const hasGlobalValue = !!currentValue;
+    const isOverridden = !deepEqual(currentValue, userGlobals[id]);
```

Comment:
a deep equal check might be problematic for cases where functions or class instances are assigned to global values, leading to potential circular references.

Why can't we check whether `userGlobals[id]` is present? This matches my expectation of an "override". Otherwise, we should also rename the variable to something like `isDeepEqual`.

```suggestion
    const isOverridden = id in userGlobals;
```

Reply:
@valentinpalkovic:

> a deep equal check might be problematic for cases where functions or class instances are assigned to global values
globals need to be serializable to go over the channel, so I don't think your examples are possible. But there may be others we should consider.

> Why can't we check whether userGlobals[id] is present? 

So I think you have this backwards (this might be a sign that the naming is bad) - `userGlobals` are the globals the user has specified via clicking toolbars etc. `globals = { ...userGlobals, ...globalOverrides }`. So this is how we check if a `globalOverride` is set for a key (does `globals[key] == userGlobals[key]`).

Now I think about it, maybe we've made it difficult to do the thing that you generally would want to do (know if a value has been overrdden). Perhaps instead we should set `{ globals, globalOverrides }` on the event, or even `{ globals, globalOverrides, userGlobals }`?

WDYT @shilman